provider/datagen/src/lib.rs

```diff
@@ -306,6 +306,13 @@ impl std::fmt::Display for CollationHanDatabase {
 }
 
 /// A language's CLDR coverage level.
+///
+/// Note that in CLDR usage, these levels are understood to be additive, i.e. the "basic"
+/// level includes all languages that are either "basic", "moderate", or "modern". In ICU4X on the
+/// other hand, a language is only part of a single coverage level. This allows generating different
+/// data files for different coverage levels without duplicating data. However, the data itself is
+/// still additive (e.g. for fallback to work correctly), so data for moderate (basic)
+/// languages should only be loaded if modern (modern and moderate) data is already present.
```

Comment:
Nit: I prefer to put positive example up front and contrasting example behind it because people sometimes only read the first sentence

```suggestion
/// In ICU4X, these are disjoint sets: a language belongs to a single coverage level. This allows
/// generating different data files for different coverage levels without duplicating data. This
/// contrasts with CLDR usage, where these levels are understood to be additive (i.e., "basic"
/// implying "basic", "moderate", and "modern"). However, the ICU4X data itself is
/// still additive (e.g. for fallback to work correctly), so data for moderate (basic)
/// languages should only be loaded if modern (modern and moderate) data is already present.
```